libraries/core_libs/storage/src/storage.cpp

```diff
@@ -1122,6 +1136,22 @@ std::vector<std::shared_ptr<DagBlock>> DbStorage::getFinalizedDagBlockByPeriod(P
   return ret;
 }
 
+std::pair<blk_hash_t, std::vector<std::shared_ptr<DagBlock>>>
+DbStorage::getLastPbftblockHashAndFinalizedDagBlockByPeriod(PbftPeriod period) {
+  std::vector<std::shared_ptr<DagBlock>> ret;
+  blk_hash_t last_pbft_block_hash;
+  if (auto period_data = getPeriodDataRaw(period); period_data.size() > 0) {
+    auto dag_blocks_data = dev::RLP(period_data)[DAG_BLOCKS_POS_IN_PERIOD_DATA];
```

Comment:
I think its better to move RLP to a separate variable, so you can reuse it below to get pbft block hash

Reply:
Fixed